extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -389,20 +392,62 @@ export class AmpStoryPage extends AMP.BaseElement {
     if (autoAdvanceAttr !== null || storyNextUpParam === null) {
       return;
     }
+    addAttributesToElement(
+      this.element,
+      dict({
+        'auto-advance-after': storyNextUpParam,
+      })
+    );
+    this.listenAndUpdateAutoAdvanceDuration_();
+  }
+
+  /**
+   * If there's a video on the page, this sets a listener to update
+   * the TimeBasedAdvancement when the first video's duration becomes available.
+   * @private
+   */
+  listenAndUpdateAutoAdvanceDuration_() {
     const video = this.getFirstAmpVideo_();
     if (video === null) {
-      addAttributesToElement(
-        this.element,
-        dict({'auto-advance-after': storyNextUpParam})
-      );
       return;
     }
-    if (video.id === '') {
-      video.id = this.element.id + '_firstVideo';
+    whenUpgradedToCustomElement(video)
+      .then(() => {
+        return video.getImpl();
+      })
```

Comment:
super nit: you can use one line notation in an arrow function. It will return the value without needing the `return` keyword.

```suggestion
      .then(() => video.getImpl())
```